### PR TITLE
UNR-2209 Bugfix/show invalid deploy config in console

### PR DESCRIPTION
**Pull request by Joe-Improbable**
#### Description
When using the deploy from editor feature having an invalid assembly name (names with hypens, etc) only shows the notification "Deployment config is not valid"

This has been changed so when validating the config any validation errors will be printed into the output log. 
The cloud deployment modal has been changed to a window, so the output log is updated whilst it's open.

#### Release note
Bugfix: Fixed an issue where launching a cloud deployment with an invalid assembly name or deployment name wouldn't show a helpful error message 

#### Tests
Tested with valid and invalid assembly and deployment names

#### Documentation
None

#### Primary reviewers
@joshuahuburn 
@improbable-valentyn

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp [hunk 1]
```diff
@@ -100,31 +101,43 @@ bool USpatialGDKEditorSettings::IsAssemblyNameValid(const FString& Name)
 {
 	const FRegexPattern AssemblyPatternRegex(SpatialConstants::AssemblyPattern);
 	FRegexMatcher RegMatcher(AssemblyPatternRegex, Name);
-
-	return RegMatcher.FindNext();
+	bool valid = RegMatcher.FindNext();
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]:
```suggestion
	bool bValidAssemblyName = RegMatcher.FindNext();
```

Comment (joshuahuburn) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]: Generally we use `UpperCamelCase` for our variable names. Booleans start with `b` though.

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp [hunk 2]
```diff
@@ -100,31 +101,43 @@ bool USpatialGDKEditorSettings::IsAssemblyNameValid(const FString& Name)
...
+	if (!valid) {
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]:
```suggestion
	if (!bValidAssemblyName ) 
	{
```

Comment (joshuahuburn) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]: `{` on newlines

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp [hunk 3]
```diff
@@ -100,31 +101,43 @@ bool USpatialGDKEditorSettings::IsAssemblyNameValid(const FString& Name)
...
+		UE_LOG(LogSpatialEditorSettings, Error, TEXT("Assembly name is invalid. It should match the regex: %s"), *SpatialConstants::AssemblyPattern);
+	}
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]:
```suggestion
	}

```

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp [hunk 4]
```diff
@@ -100,31 +101,43 @@ bool USpatialGDKEditorSettings::IsAssemblyNameValid(const FString& Name)
...
+	return valid;
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]:
```suggestion
	return bValidAssemblyName;
```

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp [hunk 5]
```diff
@@ -100,31 +101,43 @@ bool USpatialGDKEditorSettings::IsAssemblyNameValid(const FString& Name)
...
 }
 
 bool USpatialGDKEditorSettings::IsProjectNameValid(const FString& Name)
 {
 	const FRegexPattern ProjectPatternRegex(SpatialConstants::ProjectPattern);
 	FRegexMatcher RegMatcher(ProjectPatternRegex, Name);
-
-	return RegMatcher.FindNext();
+	bool valid = RegMatcher.FindNext();
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]:
```suggestion
	bool bValidProjectName = RegMatcher.FindNext();
```

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp [hunk 6]
```diff
@@ -100,31 +101,43 @@ bool USpatialGDKEditorSettings::IsAssemblyNameValid(const FString& Name)
...
+	if (!valid) {
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]:
```suggestion
	if (!bValidProjectName) 
	{
```

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp [hunk 7]
```diff
@@ -100,31 +101,43 @@ bool USpatialGDKEditorSettings::IsAssemblyNameValid(const FString& Name)
...
+		UE_LOG(LogSpatialEditorSettings, Error, TEXT("Project name is invalid. It should match the regex: %s"), *SpatialConstants::ProjectPattern);
+	}
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]:
```suggestion
	}

```

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp [hunk 8]
```diff
@@ -100,31 +101,43 @@ bool USpatialGDKEditorSettings::IsAssemblyNameValid(const FString& Name)
...
+	return valid;
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]:
```suggestion
	return bValidProjectName;
```

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp [hunk 9]
```diff
@@ -100,31 +101,43 @@ bool USpatialGDKEditorSettings::IsAssemblyNameValid(const FString& Name)
...
 }
 
 bool USpatialGDKEditorSettings::IsDeploymentNameValid(const FString& Name)
 {
 	const FRegexPattern DeploymentPatternRegex(SpatialConstants::DeploymentPattern);
 	FRegexMatcher RegMatcher(DeploymentPatternRegex, Name);
-
-	return RegMatcher.FindNext();
+	bool valid = RegMatcher.FindNext();
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]:
```suggestion
	bool bValidDeploymentName = RegMatcher.FindNext();
```

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp [hunk 10]
```diff
@@ -100,31 +101,43 @@ bool USpatialGDKEditorSettings::IsAssemblyNameValid(const FString& Name)
...
+	if (!valid) {
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]:
```suggestion
	if (!bValidDeploymentName)
	{
```

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp [hunk 11]
```diff
@@ -100,31 +101,43 @@ bool USpatialGDKEditorSettings::IsAssemblyNameValid(const FString& Name)
...
+		UE_LOG(LogSpatialEditorSettings, Error, TEXT("Assembly name is invalid. It should match the regex: %s"), *SpatialConstants::DeploymentPattern);
+	}
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]:
```suggestion
	}

```

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp [hunk 12]
```diff
@@ -100,31 +101,43 @@ bool USpatialGDKEditorSettings::IsAssemblyNameValid(const FString& Name)
 {
 	const FRegexPattern AssemblyPatternRegex(SpatialConstants::AssemblyPattern);
 	FRegexMatcher RegMatcher(AssemblyPatternRegex, Name);
-
-	return RegMatcher.FindNext();
+	bool valid = RegMatcher.FindNext();
+	if (!valid) {
+		UE_LOG(LogSpatialEditorSettings, Error, TEXT("Assembly name is invalid. It should match the regex: %s"), *SpatialConstants::AssemblyPattern);
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]: Actually, these errors should be in the call site and not in this method. It's not an error for the method to return a false result of the check, it's an error that in the call site we require a pass to continue.

Comment (joshuahuburn) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorSettings.cpp]: I suggest making this change for all the methods

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp [hunk 13]
```diff
@@ -784,8 +784,7 @@ void FSpatialGDKEditorToolbarModule::ShowSimulatedPlayerDeploymentDialog()
 	);
 
 	TSharedPtr<SWindow> RootWindow = FGlobalTabmanager::Get()->GetRootWindow();
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]:
```suggestion
```

**Pull request merged**